### DISCUSSION TITLE
Does this work with Ubuntu?

### DISCUSSION THREAD

**Discussion by WhyIsntTouhouBuntuRealYet** [ptitSeb/box86#352] [Q&A]
Basically, I want to know if I can use this with Wine on Ubuntu for RPI 400 for the lulz. Is this possible?

**Comment by ptitSeb** [comment]
I guess Ubuntu on PI is 64bits, so you'll need armhf multiarch installation.

Once you have 32bits , than yes, no problem.

**Reply by WhyIsntTouhouBuntuRealYet** [reply]
How do I get 32 bit? And once I do, can I use both box86 and wine?

**Reply by Itai-Nelken** [reply]
see this: https://github.com/ptitSeb/box86/discussions/352#discussioncomment-439446 and yes, you can use wine but you have to install it manually, see here for more details: https://github.com/novaspirit/rpi_ubu64_desktop

**Reply by WhyIsntTouhouBuntuRealYet** [reply]
I keep getting random errors, and I don't know why.
Can you provide step-by-step instructions on how to install box86 and wine starting with a fresh reinstall of Ubuntu 20.10?

**Reply by Itai-Nelken** [reply]
@WhyIsntTouhouBuntuRealYet here:
## Ubuntu 20.10:
### box86
```bash
cd $HOME
sudo dpkg --add-architecture armhf
sudo apt update
sudo apt install libc6:armhf  libx11-6:armhf  libgdk-pixbuf2.0-0:armhf libgtk2.0-0:armhf libstdc++6:armhf libsdl2-2.0-0:armhf mesa-va-drivers:armhf libsdl1.2-dev:armhf libsdl-mixer1.2:armhf libpng16-16:armhf libcal3d12v5:armhf libsdl2-net-2.0-0:armhf libopenal1:armhf libsdl2-image-2.0-0:armhf libvorbis-dev:armhf libcurl4:armhf osspd:armhf pulseaudio:armhf libjpeg62:armhf libudev1:armhf libgl1-mesa-dev:armhf libsnappy1v5:armhf libx11-dev:armhf libsmpeg0:armhf libboost-filesystem1.71.0:armhf libboost-program-options1.71.0:armhf libavcodec58:armhf libavformat58:armhf libswscale5:armhf libmyguiengine3debian1v5:armhf libboost-iostreams1.71.0:armhf  libsdl2-mixer-2.0-0:armhf
git clone https://github.com/ptitSeb/box86
cd box86
sudo apt update
sudo apt install gcc-arm-linux-gnueabihf
mkdir build; cd build; cmake .. -DRK3399=1 -DCMAKE_BUILD_TYPE=RelWithDebInfo; make -j4
sudo make install
sudo systemctl restart systemd-binfmt
```
I'm not sure about wine, but try this, it should work:
```bash
cd $HOME
sudo apt install cabextract libncurses5:armhf -y
wget https://twisteros.com/wine.tgz -O ~/wine.tgz
tar zxf ~/wine.tgz
rm ~/wine.tgz
sudo ln -s ~/wine/bin/winecfg /usr/local/bin/winecfg
sudo ln -s ~/wine/bin/wineserver /usr/local/bin/wineserver
sudo ln -s ~/wine/bin/wineboot /usr/local/bin/wineboot
echo "#!/bin/bash
setarch linux32 -L $HOME/wine/bin/wine"' "$@"' | sudo tee /usr/local/bin/wine >/dev/null
sudo chmod +x /usr/local/bin/wine /usr/local/bin/wineboot /usr/local/bin/wineserver /usr/local/bin/winecfg
sudo wget https://raw.githubusercontent.com/Winetricks/winetricks/master/src/winetricks -O /usr/local/bin/winetricks
sudo chmod +x /usr/local/bin/winetricks
```
run `wine wineboot` to create a new wine prefix.
run
```bash
sudo sed -i 's|echo "\${arg%%=\*}"=\\""${arg### \*=}"\\"|echo \${arg%%=\*}=\\"\${arg### \*=}\\"|g' /usr/local/bin/winetricks
```
if you get adobeair warning when running winetricks.


**sources:**
https://itai-nelken.github.io/Ubuntu-on-rpi-fixes/
https://github.com/novaspirit/rpi_ubu64_desktop

**Reply by WhyIsntTouhouBuntuRealYet** [reply]
Thanks. I'll let you know if it works.

**Reply by WhyIsntTouhouBuntuRealYet** [reply]
It works. I had to install wine normally, as well as other things. Only issue is that the sound cuts out when I play a game like Touhou, but other than that, it plays fine.

**Reply by Pythonic456** [reply]
For a possibly easier way to install wine+box86, you could try [Pi-Apps](https://github.com/Botspot/pi-apps).

Of course, this will have to be done in armhf.

**Reply by Itai-Nelken** [reply]
> For a possibly easier way to install wine+box86, you could try [Pi-Apps](https://github.com/Botspot/pi-apps).
> 
> Of course, this will have to be done in armhf.

me and Botspot are working on getting box86 on arm64 to pi-apps (using multiarch), so that means wine for arm64 might be added to pi-apps!

**Comment by Itai-Nelken** [comment]
I have a way to install it here: https://itai-nelken.github.io/Ubuntu-on-rpi-fixes/
